### ISSUE TITLE
Review content traffic

### ISSUE THREAD

**Issue by Rick-Anderson** [aspnet/Docs#7070]
Review traffic

* Getting the traffic we expect.
*  Topics with high bounce rate because of landing on the wrong place
* CSAT on Top 20. Look at CSAT on top 100 and see if there's a pattern - why is CSAT low.
* What does the traffic trend look like for the whole area over the last two years?  What does that trend look like for conceptual, for API Ref?
* Any action to take based on what you see in the data?

**Comment by guardrex** [comment]
> CSAT on Top 20. Look at CSAT on top 100 and see if there's a pattern - why is CSAT low.

### TL;DR

Consider a new stat, **Customer Negative Sentiment (CNS)** for assessing topics for major work that focuses on the negative review clicks and the unique topic visitor count (less bounces).

### Discussion

#### CSAT can be misleading

I think using CSAT alone can be misleading. I suggest that the number of unique visitors (less bounces) is important and that positive and total review clicks aren't as important. I think a general assessment stat for topics should lean more in the direction of a *negative sentiment percentage* with a *cutoff for action*. Such a statistic may provide a better handle on which topics require major work.

Let's assume no bounces for the following comparison:

Twenty negative review clicks out of 40 reviews (CSAT = 50%) would be horri-terrible if there were 40 unique visitors to a page. Truly, 50% of the readers are clearly upset! Sweeping changes to the scope, depth, and layout of the content should be considered.

Twenty negative review clicks out of 40 reviews for a topic that had 20,000 visitors is a **_great result_**. Only 0.1% of the readership is obviously upset. That's such a small fraction of the readership that I don't think one can make an argument for major topic changes. Attempting to please the 20 upset readers with a topic overhaul/rewrite runs the risk of displeasing a larger number of the 19,960 readers who didn't click. The number of negative reviews is **at risk of increasing** if sweeping changes are made.

#### Supremacy of comments

Even if a topic has one negative review and 50 positive reviews, the one negative review might be a *bug* in the topic that must be addressed. Although 20 negative reviews out of 20,000 visitors shouldn't prompt major changes, those 20 reviews may include comments on specific points that require attention. From speaking with other authors, I think we agree on this point: The comments are more helpful than the broad stats.

#### Role of positive reviews

The positive reviews and the non-responses might be conflated. When I read a topic and I find what I want, I usually just go back to work with the info. I don't go out of my way to tell anyone I'm happy. I feel that someone was paid to do their job. The topic was supposed to explain the tech, it did, and I need to get back to work.

I wish I had more knowledge on survey statistics because I think this aspect is an understood subject in that field. Dividing positive/negative reviews by total reviews is probably a useful approach when one has a good understanding of how likely readers click when they're happy with the topic's coverage. If that fraction is low relative to negative clicks or the relationship is unknown, I think a case can be made for dropping positive and total reviews from the analysis and focusing on the negative clicks.

### Conclusion & Proposal

There are alternatives for CSAT: Divide it by unique visitors. Perhaps, divide it by the highest CSAT topic in the repo. In my opinion, these approaches are unnecessarily complex to interpret. A simple negative sentiment index with a cutoff is easy to understand and apply.

I pitch that the number of negative reviews be divided by the unique visitor count less the page's bounces:

**Customer Negative Sentiment (CNS) = Negative Clicks / (Unique Visitors - Bounces)**

The cutoff for an overhaul is probably in the 5-10% range because so many upset readers won't click or find the button. For every 100 readers of a topic where more than 5 readers found the button and clicked a negative review, that topic is probably a candidate for major changes. When that number goes to 10+ readers out of 100, then I suspect there's a need for a full overhaul/rewrite of the topic. We can debate the numbers here, but it seems much easier to me (and valid) to adopt this approach.

If it can be shown that there is a high inverse correlation between CSAT and CNS, that would bolster the usefulness of CSAT. I predict ... I might be wrong ... but I predict that the correlation won't be so hot. I'd need to run the full dataset to check. If you'd like to send me the data, I'd be happy to perform the analysis. I would need the unique visitors, negative click count, bounces, and CSAT for all of the topics on the repo.

@Rick-Anderson @rachelappel @scottaddie Since we know the [File Uploads topic](https://docs.microsoft.com/aspnet/core/mvc/models/file-uploads) receives complaints in feedback issues (#5331 AFAIK the sample is actually broken) and we probably agree on a rewrite for that topic, it might be helpful to see that topic's stats, including the bounces. Looking at one topic is anecdotal for broad conclusions, but that's one where we know readers are very unhappy. Would you mind at least posting all of the stats for that one (or send them to me for analysis)?

cc/ @wadepickett

**Comment by Rick-Anderson** [comment]
Done
